web/src/compose_validate.js

```diff
@@ -118,10 +131,6 @@ export function warn_if_private_stream_is_linked(linked_stream, $textarea) {
 }
 
 export function warn_if_mentioning_unsubscribed_user(mentioned, $textarea) {
-    if (compose_state.get_message_type() !== "stream") {
```

Comment:
If we mention someone in DM, then we won't receive the `stream_id` and will make an early return.